gapis/api/gles/gles.api

```diff
@@ -570,7 +570,25 @@ class Context {
 // Globals
 ContextID              NextContextID
 map!(u64, ref!Context) Contexts
-ref!Context            CurrentContext
+
+// Internal "thread-local" global for quick access the current context.
+// It is set in the PreMutate() function before any command is mutated.
+ref!Context CurrentContext
+
+// Internal sub-routine which is called before any command is mutated.
+// TODO: The IsGLESCommand() is a hack. We should move all EGL commands to own api.
+// TODO: We should also set the Version and Extension global variables here.
+// TODO: Many commands have @if annotation. We evaluate it and check it here.
+sub void PreMutate() {
+  CurrentContext = Contexts[$Thread]
```

Comment:
As this method is never referenced directly by the `.api` file the static analysis will consider this function dead code. All static analysis then becomes heavily dependent on the single assignment to `CurrentContext` in `SetContext`. Without static analysis, things like enum inference will likely break.

This seems really fragile. If we are to really go this way, `PreMutate` should become a first-class language concept. My personal feeling is that we should just call the function at the top of all GL functions, and remove all the hacky GL vs EGL stuff from GAPIS and GAPII.

Reply:
The primary purpose is to ensure we match behavior between gapii and gapis.
Secondary on is to allow cleaning of the language.  Calling at the start of every command is messy IMFO.
Having said that, this commit is not important, I just picked it from the other WIP PR as it seems like decent improvement.  If you do not like it, I will rather just drop it.

Reply:
Please let's remove this CL from this PR.
I only really want to be landing changes that are bug fixes or critical at this point. This is neither.

Reply:
Already did